### ISSUE TITLE
Keep installed pip packages even when notebook is terminated

### ISSUE THREAD

**Issue by yinweisu** [awslabs/aws-orbit-workbench#440]
Issue: 
Currently installed pip packages will be gone if the notebook is terminated. This could be an issue as the notebook will be terminated if it's not active, and users will have to reinstall everything again.

Tried Solution:
We tried to change the default pip installation path to be a folder in EFS by setting env variables in Dockerfile, so that packages won't be removed. However, it looks like the folder in EFS got recreated every time the notebook boots up. Below is a snippet of code we used:
```
# Customize pip installation location
RUN mkdir -p /home/jovyan/private/site_packages
ENV PIP_TARGET=/home/jovyan/private/site_packages
ENV PYTHONPATH=$PYTHONPATH:/home/jovyan/private/site_packages
```

Question:
Isn't everything in private permanent and won't be removed after each launch up? Or is there any special setting in jupyterhub that disable us doing it. Also, is there any better solution to address this pip package issue?

**Comment by abaror** [comment]
everything in private is permanent .  As part of the kubeflow changes, we will move to a state where the entire home directory is in efs under the user home dir.  Then we can test this again.

**Comment by stthoom** [comment]
Adding fsx location to conda envs paths, cloning env and adding to python env to kernel using below commands. 


conda config  --prepend envs_dirs  /fsx/condaenvs
conda create --clone base --name conda_env_clone
conda activate conda_env_clone
conda install -c anaconda ipykernel
python -m ipykernel install --user --name conda_env_clone --display-name “Python (conda_env_clone)”


Worked from vscode terminal but from  JL notebook/terminal activating conda environment throws below error even after doing conda init and opening new terminal

++ conda activate conda_env_clone

CommandNotFoundError: Your shell has not been properly configured to use 'conda activate'.
To initialize your shell, run

    $ conda init <SHELL_NAME>

Currently supported shells are:
  - bash
  - fish
  - tcsh
  - xonsh
  - zsh
  - powershell

See 'conda init --help' for more information and options.

IMPORTANT: You may need to close and restart your shell after running 'conda init'.

**Comment by stthoom** [comment]
https://awslabs.github.io/aws-orbit-workbench/reusable-conda-environments

Need to fix the documentation site 404.

**Comment by stthoom** [comment]
https://github.com/awslabs/aws-orbit-workbench/blob/gh-pages/tutorials/reusable-conda-environments.markdown

As part of testing, identified the kernel selection is not shifting the python environment.  Per configurations can see file paths int he kernet config file. Currently debugging the issue. 


!cat /home/jovyan/.local/share/jupyter/kernels/conda_env_clone_example/kernel.json

{
 "argv": [
  "/fsx/condaenvs/conda_env_clone_example/bin/python",
  "-m",
  "ipykernel_launcher",
  "-f",
  "{connection_file}"
 ],
 "display_name": "Python (conda_env_clone_example)",
 "language": "python"
}

**Comment by stthoom** [comment]
https://github.com/awslabs/aws-orbit-workbench/pull/700/files

Waiting for merge.

**Comment by yinweisu** [comment]
Hi, 

Just noticed two issues while following the guide: https://awslabs.github.io/aws-orbit-workbench/reusable-conda-environments
1.  Users don't have permission to write to fsx when cloning the env. I had to use sudo.
2.  I created a conda env and added that to jupyter kernel in notebook server A. I was able to see the kernel in notebook server B too, but using it would lead to error:
![image](https://user-images.githubusercontent.com/21029745/128939754-38a85153-3bab-4337-b621-842ffe1d7e01.png)
